core/src/bmi.jl

```diff
@@ -16,6 +16,10 @@ function BMI.initialize(T::Type{Model}, config::Config)::Model
     local parameters, state, n
     try
         parameters = Parameters(db, config)
+        if !valid_flow_neighbor_amounts(parameters)
+            error("Invalid neighbor amounts.")
```

Comment:
I think "amounts" isn't the appropriate when we're talking about the number of neighbors. (Amount generally refers to uncountable entities.)

This is also a little terse (although I guess the error message should make enough sense will everything that has been logged via @error).

I would rephrase to something like "Invalid network definition:" so it acts as a header and then the content of errors will clarify it.